go/tools/builders/compilepkg.go

```diff
@@ -270,18 +271,24 @@ func compileArchive(
 	// constraints.
 	compilingWithCgo := haveCgo && cgoEnabled
 
+	filterForNogo := func(slice []string) []string {
+		var filtered []string
```

Comment:
optional, pre-allocate
```suggestion
		filtered := make([]string, 0, len(slice))
```